src/neptune/new/handler.py

```diff
@@ -253,43 +257,120 @@ def log(
-                else:
-                    first_value = value
-
-                if is_float(first_value):
-                    attr = FloatSeries(self._container, parse_path(self._path))
-                elif is_string(first_value):
-                    attr = StringSeries(self._container, parse_path(self._path))
-                elif FileVal.is_convertable(first_value):
-                    attr = FileSeries(self._container, parse_path(self._path))
-                elif is_float_like(first_value):
-                    attr = FloatSeries(self._container, parse_path(self._path))
-                elif is_string_like(first_value):
-                    warn_once(
-                        message="The object you're logging will be implicitly cast to a string."
-                        " We'll end support of this behavior in `neptune-client==1.0.0`."
-                        " To log the object as a string, use `.log(str(object))` instead.",
-                        stack_level=3,
-                    )
-                    attr = StringSeries(self._container, parse_path(self._path))
-                else:
-                    raise TypeError("Value of unsupported type {}".format(type(first_value)))
-
+                    raise ValueError("Value cannot be a collection")
+                attr = self._create_new_attribute(value)
                 self._container.set_attribute(self._path, attr)
             attr.log(value, step=step, timestamp=timestamp, wait=wait, **kwargs)
 
+    def _create_new_attribute(self, value):
+        if is_float(value):
+            return FloatSeries(self._container, parse_path(self._path))
+        elif is_dict_like(value):
+            return Namespace(self._container, parse_path(self._path))
+        elif is_string(value):
+            return StringSeries(self._container, parse_path(self._path))
+        elif FileVal.is_convertable(value):
+            return FileSeries(self._container, parse_path(self._path))
+        elif is_float_like(value):
+            return FloatSeries(self._container, parse_path(self._path))
+        elif is_string_like(value):
+            warn_once(
+                message="The object you're logging will be implicitly cast to a string."
+                " We'll end support of this behavior in `neptune-client==1.0.0`."
+                " To log the object as a string, use `.log(str(object))` instead.",
+                stack_level=3,
+            )
+            return StringSeries(self._container, parse_path(self._path))
+        else:
+            raise TypeError("Value of unsupported type {}".format(type(value)))
+
+    def extend(
+        self,
+        values: Union[Dict[str, Iterable[Any]], Iterable[Any]],
+        steps: Optional[Iterable[float]] = None,
+        timestamps: Optional[Iterable[float]] = None,
+        wait: bool = False,
+        **kwargs,
+    ) -> None:
+        """
+        todo: NPT-12530
+        """
+        if is_dict_like(values):
```

Comment:
Shuldn't we just raise an exception that user tried to `extend` dict value instead of collection?

Reply:
user can extend dict, but dict must contain Iterables. However i need to add exception when user will extend Value object instead of dict or iterable.